update MenuItem to not support click and its not a link if disabled (#1281)
looker-open-source/components: www/src/documentation/components/overlays/menu.mdx

`MenuItem` is an `HTMLButtonElement` that supports all of the `BoxProps` properties. Use this for triggering actions from with in a `MenuList`. For example, opening a dialog.

For accessibility reasons, if you want your `MenuItem` to link somewhere then you can use the `itemRole` prop to identify it as a link.

> **Note:** If using the props `itemRole` don't use the prop `disabled` As it is bad-form and defeat the purpose.

Reviewer: ```suggestion

**Note:**  `itemRole="link"` and `disabled` cannot be combined (use `itemRole="button" if you need to offer a disabled `MenuItem`)
```

Author: thanks I knew that wasn't quite right but wan't sure what should be!
